<?xml version="1.0"?>
<launch>
       
     <arg name="camera_model"         default="BW1098OAK" /> <!-- 'zed' or 'zedm' -->
    <arg name="camera_name"          default="oak" />

    <arg name="base_frame"           default="oak-d_frame" />

    <arg name="cam_pos_x"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" /> <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <!-- args for stereo node -->
    <!-- <arg name="outputDepth"           default="true" /> 
    <arg name="outputRectified"       default="true" /> 
    <arg name="lrcheck"               default="true" /> 
    <arg name="extended"              default="false" /> 
    <arg name="subpixel"              default="false" />  -->
    <arg name="camera_param_uri"      default="package://depthai_examples/params/camera" />

   
    <include file="$(find depthai_bridge)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="camera_name"     value="$(arg  camera_name)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"   />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"   />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"   />
        <arg name="cam_roll"        value="$(arg  cam_roll)"    />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"   />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"     />
    </include>


    <node name="stereo_publisher" pkg="depthai_examples" type="stereo_node" output="screen" required="true">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>
    </node>            


    <!-- ~~~~~~~~~~~~~~~~~~~~~~~~  depth_image_proc nodelet below ~~~~~~~~~~~~~~~~~~~ -->

    <node pkg="nodelet" type="nodelet" name="nodelet_manager"  args="manager" output="screen"/>
    
    <!-- <node pkg="nodelet" type="nodelet" name="nodelet_manager"  args="manager" output="screen">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>
    </node> -->


    <node pkg="nodelet" type="nodelet" name="depth_image_convertion_nodelet"
        args="load depth_image_proc/convert_metric nodelet_manager">
        <remap from="image_raw" to="/stereo_publisher/stereo/depth"/>    
        <remap from="image" to="/image_proc/stereo/image"/>
    </node>


    <!-- <node pkg="nodelet" type="nodelet" name="depth_image_to_pointcloud_nodelet"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
        <remap from="camera_info" to="/stereo_publisher/stereo/camera_info"/>
        <remap from="image_rect" to="/image_proc/stereo/image"/>    
        <remap from="points" to="/stereo/points"/>
    </node> -->

    <!-- Rviz configuration -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find depthai_examples)/rviz/stereo_nodelet.rviz" />

</launch>
